packages/dd-trace/src/appsec/iast/analyzers/hsts-header-missing-analyzer.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const { HSTS_HEADER_MISSING } = require('../vulnerabilities')
+const { MissingHeaderAnalyzer } = require('./missing-header-analyzer')
+
+const HSTS_HEADER_NAME = 'Strict-Transport-Security'
+const HEADER_VALID_PREFIX = 'max-age'
+class HstsHeaderMissingAnalyzer extends MissingHeaderAnalyzer {
+  constructor () {
+    super(HSTS_HEADER_MISSING, HSTS_HEADER_NAME)
+  }
+  _isVulnerableFromRequestAndResponse (req, res) {
+    const headerToCheck = res.getHeader(HSTS_HEADER_NAME)
+    return !this._isHeaderValid(headerToCheck) && this._isHttpsProtocol(req)
+  }
+
+  _isHeaderValid (headerValue) {
+    if (!headerValue) {
+      return false
+    }
+    headerValue = headerValue.trim()
+
+    if (!headerValue.startsWith(HEADER_VALID_PREFIX)) {
+      return false
+    }
+
+    const semicolonIndex = headerValue.indexOf(';')
+    let timestampString
+    if (semicolonIndex > -1) {
+      timestampString = headerValue.substring(HEADER_VALID_PREFIX.length + 1, semicolonIndex)
+    } else {
+      timestampString = headerValue.substring(HEADER_VALID_PREFIX.length + 1)
+    }
+
+    const timestamp = parseInt(timestampString)
+    // eslint-disable-next-line eqeqeq
+    return timestamp == timestampString && timestamp > 0
+  }
+
+  _isHttpsProtocol (req) {
+    return req.protocol === 'https' || req.headers['x-forwarded-proto'] === 'https'
```

Comment:
`protocol` property is only present in `express`. For raw `http(s)` server it will always be false.

Reply:
you are right, that is the reason that we are cheking also `x-forwarded-proto` header.

Reply:
I was thinking in the case the server was created with `https` module.

Reply:
yes, we can assume that, we are doing our best effort to try to get a protocol, if we can't we are not going to detect this vulnerability, because it only can happen in https.